cachi2/core/scm.py

```diff
@@ -20,32 +20,42 @@ def clone_as_tarball(url: str, ref: str, to_path: Path) -> None:
     :param ref: the revision to check out
     :param to_path: create the tarball at this path
     """
+    list_url = [url]
+    # Fallback to `https` if cloning source via ssh fails
+    if "ssh://" in url:
+        list_url.append(url.replace("ssh://", "https://"))
+
     with tempfile.TemporaryDirectory(prefix="cachito-") as temp_dir:
-        log.debug("Cloning the Git repository from %s", url)
-        try:
-            repo = git.repo.Repo.clone_from(
-                url,
-                temp_dir,
-                no_checkout=True,
-                filter="blob:none",
-                # Don't allow git to prompt for a username if we don't have access
-                env={"GIT_TERMINAL_PROMPT": "0"},
-            )
-        except Exception as ex:
-            log.exception(
-                "Failed cloning the Git repository from %s, ref: %s, exception: %s",
-                url,
-                ref,
-                type(ex).__name__,
-            )
-            raise FetchError("Failed cloning the Git repository")
+        for url in list_url:
+            log.debug("Cloning the Git repository from %s", url)
+            try:
+                repo = git.repo.Repo.clone_from(
+                    url,
+                    temp_dir,
+                    no_checkout=True,
+                    filter="blob:none",
+                    # Don't allow git to prompt for a username if we don't have access
+                    env={"GIT_TERMINAL_PROMPT": "0"},
+                )
+            except Exception as ex:
+                log.warning(
+                    "Failed cloning the Git repository from %s, ref: %s, exception: %s",
+                    url,
+                    ref,
+                    type(ex).__name__,
+                )
+                continue
```

Comment:
There's not enough info here to determine what happened. Previously, because we used `log.exception`, we had the whole traceback, but now we have only the exception class name. `str(ex)` should hopefully have some details

Reply:
should I file a follow up story for this warning message top get more info?

Reply:
```
2023-05-26 09:05:50,452 WARNING Failed cloning the Git repository from ssh://git@bitbucket.org/cachi-testing/cachi2-without-deps.git, ref: 9e164b97043a2d91bbeb992f6cc68a3d1015086a, exception: GitCommandError, exception-msg: Cmd('git') failed due to: exit code(128)
  cmdline: git clone -v --no-checkout --filter=blob:none -- ssh://*****@bitbucket.org/cachi-testing/cachi2-without-deps.git /tmp/cachito-zobvvxxs
  stderr: 'Cloning into '/tmp/cachito-zobvvxxs'...
Host key verification failed.
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
'
2023-05-26 09:05:51,885 WARNING Failed cloning the Git repository from ssh://git@bitbucket.org/cachi-testing/cachi2-without-deps-second.git, ref: 09992d418fc44a2895b7a9ff27c4e32d6f74a982, exception: GitCommandError, exception-msg: Cmd('git') failed due to: exit code(128)
  cmdline: git clone -v --no-checkout --filter=blob:none -- ssh://*****@bitbucket.org/cachi-testing/cachi2-without-deps-second.git /tmp/cachito-fno7_6cc
  stderr: 'Cloning into '/tmp/cachito-fno7_6cc'...
Host key verification failed.
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
'

```

Reply:
Looks better now.